### PR TITLE
Add localAddress() and remoteAddress()

### PR DESCRIPTION
**Pull request by LuKks**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -109,6 +109,30 @@ class DHT extends EventEmitter {
     return socket ? socket.address() : null
   }
 
+  localAddress () {
+    if (!this.io.serverSocket) return null
```
Comment (LuKks) [inline index.js:113]: This line and another line are the only differences between the methods found on `hyperdht`

File: index.js [hunk 2]
```diff
@@ -109,6 +109,30 @@ class DHT extends EventEmitter {
...
+      host: localIP(this.udx),
+      port: this.io.serverSocket.address().port
+    }
+  }
+
+  remoteAddress () {
+    if (!this.host) return null
+    if (!this.port) return null
+    if (this.firewalled) return null
+    if (!this.io.serverSocket) return null
```
Comment (LuKks) [inline index.js:125]: This is the other line

File: test.js [hunk 3]
```diff
@@ -454,6 +454,69 @@ test('close event is only emitted once', async function (t) {
...
 })
 
+test('local address', async function (t) {
+  t.plan(4)
+
+  const node = new DHT()
+  t.is(node.localAddress(), null)
+
+  await node.ready()
+  t.is(node.localAddress().host, localIP(4))
```
Comment (mafintosh) [inline test.js:464]: instead of copying the impl, just test that it looks like an ip, otherwise what are we testing 🤣

Comment (LuKks) [reply test.js:464]: Haha true!

**Pull request merged**